packages/framework/tinylicious-client/src/test/TinyliciousClient.spec.ts

```diff
@@ -277,4 +279,63 @@ describe("TinyliciousClient", () => {
         await corruptedAliasOp(runtimeOf(do1), "alias");
         assert(await dataCorruption);
     });
+
+    /**
+     * Scenario: Test if TinyliciousClient with forceWriteMode set to true forces the container to write mode.
+     * Keep the original behaviour otherwise.
+     *
+     * Expected behavior: Setting forceWriteMode to true in TinyliciousClientProps should force the container's
+     * connectionMode to `write` while keeping the original behavior if forceWriteMode is not set to true.
+     */
+    it("forces the container to connect with write mode when forceWriteMode is set to true ", async () => {
+        const getConnectionModeFromTinyliciousClient = async (
+            props?: TinyliciousClientProps,
+        ): Promise<ConnectionMode> => {
+            const client = new TinyliciousClient(props);
+            const { container } = await client.createContainer(schema);
+            await container.attach();
+
+            const map1 = container.initialObjects.map1 as SharedMap;
+            map1.set("newpair-id", "test");
+
+            await timeoutPromise((resolve) => container.once("connected", resolve), {
```

Comment:
Do we really need to wait for "connected" here?  The websocket connection should already be established after awaiting the container.attach() call above, so it seems we should already know whether we got the right connection mode?

Reply:
Yeah, I looked at this briefly on Friday. It seems on container's attach, we do not wait for delta manager connection to be established before we return. I was thinking to look at the feature work around connection state, before filing this as an issue. But from the perspective of this feature work, it seems valid to listen to connected state before checking on the mode established during that connection.

Reply:
Having said that I would still ensure in the unit tests we are not moving to a write mode (basically another disconnect/connect) because we are "writing" on the map.